2_standard_libraries/itertools_examples.py

```diff
@@ -0,0 +1,36 @@
+from itertools import groupby, takewhile, dropwhile
+
+
+def do_group_by():
+    #  groupby returns consecutive keys and groups / lot's of possible uses here but too advanced now
+    my_list = [('crank', 'shimano'), ('frame', 'bmc'), ('frame', 'giant'),
+               ('crank', 'sram')]
+    for brands, components in groupby(my_list, lambda x: x[
+        0]):  # lambda tells groupby to use first item as grouping key
+        for item in components:
+            print(f'We have a {brands} by {item[1]}')
+
+
+def do_takewhile():  # ???? I don't understand this, it's not iterating and just stops at the first odd number
+
+    def is_even(x):
+        return x % 2 == 0
+
+    my_list = [2,3,4]
+    result = list(takewhile(is_even, my_list))
+    print(result)
+
+def do_dropwhile():  # display after first time the condition is false, what for???
```

Comment:
```
Make an iterator that drops elements from the iterable as long as the predicate is true
```

that's the opposite of `takewhile`.  the use cases can be different depending on the task. consider that `iterable` can be not only a list. we should get an idea of how it works.